crates/rust-analyzer/src/req.rs

```diff
@@ -216,6 +216,14 @@ impl Request for Ssr {
     const METHOD: &'static str = "rust-analyzer/ssr";
 }
 
+pub enum SsrQueryCheck {}
+
+impl Request for SsrQueryCheck {
+    type Params = SsrParams;
+    type Result = Option<String>;
+    const METHOD: &'static str = "rust-analyzer/check_ssr_query";
```

Comment:
Let's reuse existing query?

```rust
pub struct SsrParams {
    pub arg: String,
    pub validate_only: bool,
}
```

Seems like this would lead to less code duplication down the line.